sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,377 @@
+        fh : guaranteed to be ForecastingHorizon
+            The forecasting horizon with the steps ahead to to predict.
+        X : optional (default=None)
+            guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            Exogeneous time series to predict from.
+        alpha : list of float, optional (default=[0.5])
+            A list of probabilities at which quantile forecasts are computed.
+
+        Returns
+        -------
+        quantiles : pd.DataFrame
+            Column has multi-index: first level is variable name from y in fit,
+                second level being the values of alpha passed to the function.
+            Row index is fh, with additional (upper) levels equal to instance levels,
+                    from y seen in fit, if y_inner_mtype is Panel or Hierarchical.
+            Entries are quantile forecasts, for var in col index,
+                at quantile probability in second col index, for the row index.
+        """
+        unknown_exogenous, known_exogenous = self.convert_exogenous_dataset(X)
+        maximum_forecast_horizon = fh.to_relative(self.cutoff)[-1]
+        absolute_fh = fh.to_absolute(self.cutoff)
+        endogenous_quantile_predictions = self._forecaster.predict(
+            maximum_forecast_horizon,
+            past_covariates=unknown_exogenous,
+            future_covariates=known_exogenous,
+            num_samples=self.num_samples,
+        ).quantiles_df(quantiles=alpha)
+        variable_names = self._get_varnames()
+        multi_index = pd.MultiIndex.from_product(
+            [variable_names, alpha], names=["variable", "quantile"]
+        )
+        expected_index = fh.get_expected_pred_idx(self.cutoff)
+        endogenous_quantile_predictions.index = (
+            endogenous_quantile_predictions.index.astype(expected_index.dtype)
+        )
+
+        abs_idx = absolute_fh.to_pandas().astype(expected_index.dtype)
+        endogenous_quantile_predictions.columns = multi_index
+        return endogenous_quantile_predictions.loc[abs_idx]
+
+
+def _handle_input_index(dataset: pd.DataFrame) -> pd.DataFrame:
+    """Convert input dataset index to the compatible type for ``darts``.
+
+    Parameters
+    ----------
+    dataset: pandas.DataFrame
+        dataset with index to be converted
+
+    Returns
+    -------
+    pandas.DataFrame
+        converted dataset
+    """
+    if isinstance(dataset.index, (pd.DatetimeIndex, pd.RangeIndex)):
+        return dataset
+    dataset_copy = dataset.copy(deep=True)
+
+    if isinstance(dataset_copy.index, pd.PeriodIndex):
+        dataset_copy.index = dataset_copy.index.to_timestamp()
+        return dataset_copy
+
+    if _is_int64_type(dataset_copy.index):
+        dataset_copy.index = pd.RangeIndex(
+            start=dataset_copy.index.min(),
+            stop=dataset_copy.index.max() + 1,
+            step=dataset_copy.index[1] - dataset_copy.index[0],
+        )
+        return dataset_copy
+
+    return dataset_copy
```

Comment:
What are the cases where this line is reached? And does the code work in those cases?

Reply:
There was a check with  Windows runner with `3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)]`  with `pandas 2.2.2`the issue is when the `NumericIndex` is directly included in `pd.Index` class for pandas 2.